cocos/base/CCRef.cpp

```diff
@@ -111,7 +116,7 @@ void Ref::release()
     --_referenceCount;
 
 #if CC_ENABLE_SCRIPT_BINDING && CC_ENABLE_GC_FOR_NATIVE_OBJECTS
-    if (_scriptOwned && _referenceCount==1 && _rooted)
+    if (_scriptOwned && _rooted && _referenceCount==_referenceCountAtRootTime)
```

Comment:
The thing is that I don't have to unroot the obj when refCount==1.
I have to unroot it when the refCount == sameRefCount when the object was rooted.